common/scanmem.c

```diff
@@ -0,0 +1,1005 @@
+/*
+ *  Copyright (C) 2021 Cisco Systems, Inc. and/or its affiliates. All rights reserved.
+ *  Copyright (C) 2005-2010 Gianluigi Tiesi <sherpya@netfarm.it>
+ *
+ *  Authors: Gianluigi Tiesi
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ *  MA 02110-1301, USA.
+ */
+
+#include <windows.h>
+#include <tlhelp32.h>
+
+#include <psapi.h>
+#include <windns.h>
+
+#include <clamav.h>
+#include <others.h>
+
+#include "actions.h"
+#include "output.h"
+#include "exescanner.h"
+#include "scanmem.h"
+#include "clamdcom.h"
+
+typedef int (*proc_callback)(PROCESSENTRY32 ProcStruct, MODULEENTRY32 me32, void *data, struct mem_info *info);
+int sock;
+
+/* clamd helpers */
+static int chkpath(const char *path, struct mem_info *minfo)
+{
+    const struct optstruct *opt;
+    struct optstruct *clamdopts = NULL;
+
+    if ((clamdopts = optparse(optget(minfo->opts, "config-file")->strarg, 0, NULL, 1, OPT_CLAMD, 0, NULL)) == NULL) {
+        logg("!Can't parse clamd configuration file %s\n", optget(minfo->opts, "config-file")->strarg);
+    }
+
+    if (!path) {
+        return 1;
+    }
+
+    if ((opt = optget(clamdopts, "ExcludePath"))->enabled) {
+        while (opt) {
+            if (match_regex(path, opt->strarg) == 1) {
+                if (!optget(minfo->opts, "infected")->enabled)
+                    logg("~%s: Excluded\n", path);
+                return 1;
+            }
+            opt = opt->nextarg;
+        }
+    }
+    return 0;
+}
+
+/* Issues an INSTREAM command to clamd and streams the given file
+ * Returns >0 on success, 0 soft fail, -1 hard fail */
+static int send_stream(int sockd, const char *filename, struct mem_info *info)
```

Comment:
I just realized this function suffers the same copypaste issue and is nearly identical to a `send_stream()` function in `proto.c`.

Below, `connect_clamd()` also suffers the same problem.  It is mostly duplicate of [the poorly named] `dconnect()` function from `proto.c`. 

Do you think it makes sense to merge these all and put them inside of `clamdcom.c`/`.h`?